elastic/kibana
# [Obs Alert Table] Refactor alert table registration and change default columns

Closes #174239

## Summary

This PR refactors the alert table registrations and registers a new alert table for the rule details page. This PR also fixes the SLO embeddable alert table issue by opening the flyout when the user clicks on the reason.

Here is the old vs the new table:

|Old alert table|New alert table|
|---|---|
|![image](https://github.com/elastic/kibana/assets/12370520/0289dc2d-b6e8-4727-96fa-b2cf6485e8e5)|![image](https://github.com/elastic/kibana/assets/12370520/226fe529-5ee0-4f5c-9e9f-5ee782c8afe6)|

## 🧪 How to test
- Create some alerts and check the alert table on the alerts page, it should 
    - have 2 hours time range instead of 15 mins
    - show the following fields
        - Triggered
        - Duration
        - Rule name
             - on hover: it shows rule category
        - Group
        - Observed value
        - Threshold
        - Tags
- Check one of the rules page alert tables, it should not show the rule name column by default
- Create an SLO and check the related alert table embeddable
   - Clicking on the reason should show the flyout (this was the fix in this PR)
   - Everything should work as before

## What will be covered in the future PRs
- [ ] Making sure all rules set the `threshold` field
- [ ] Adding a column with difference (Needs to be discussed with ResponseOps team)
- [ ] Having source column instead of group by field (Needs to be discussed with @maciejforcone )
